File: mobile/lib/features/schedule/domain/enum/schedule_block_type.dart
enum ScheduleBlockType {
  work,
  study,
  meeting,
  exercise,
  rest,
  commute,
  personal,
  other,
}

extension ScheduleBlockTypeX on ScheduleBlockType {
  static ScheduleBlockType fromApi(String value) {
    switch (value.toUpperCase()) {
      case 'WORK':
        return ScheduleBlockType.work;
      case 'STUDY':
        return ScheduleBlockType.study;
      case 'MEETING':
        return ScheduleBlockType.meeting;
      case 'EXERCISE':
        return ScheduleBlockType.exercise;
      case 'REST':
        return ScheduleBlockType.rest;
      case 'COMMUTE':
        return ScheduleBlockType.commute;
      case 'PERSONAL':
        return ScheduleBlockType.personal;
      default:
        return ScheduleBlockType.other;
    }
  }

  String get apiValue {
    switch (this) {
      case ScheduleBlockType.work:
        return 'WORK';
      case ScheduleBlockType.study:
        return 'STUDY';
      case ScheduleBlockType.meeting:
        return 'MEETING';
      case ScheduleBlockType.exercise:
        return 'EXERCISE';
      case ScheduleBlockType.rest:
        return 'REST';
      case ScheduleBlockType.commute:
        return 'COMMUTE';
      case ScheduleBlockType.personal:
        return 'PERSONAL';
      case ScheduleBlockType.other:
        return 'OTHER';
    }
  }

  String get label {
    switch (this) {
      case ScheduleBlockType.work:
        return 'Work';
      case ScheduleBlockType.study:
        return 'Study';
      case ScheduleBlockType.meeting:
        return 'Meeting';
      case ScheduleBlockType.exercise:
        return 'Exercise';
      case ScheduleBlockType.rest:
        return 'Rest';
      case ScheduleBlockType.commute:
        return 'Commute';
      case ScheduleBlockType.personal:
        return 'Personal';
      case ScheduleBlockType.other:
        return 'Other';
    }
  }
}

File: backend/src/main/java/com/lifeos/backend/schedule/domain/enums/ScheduleBlockType.java
package com.lifeos.backend.schedule.domain.enums;

/**
 * Type/category of a schedule block.
 *
 * This helps Today, Timeline, color UI, analytics, and future AI suggestions.
 */
public enum ScheduleBlockType {

    /**
     * Deep work / coding / focused execution.
     */
    DEEP_WORK,

    /**
     * Study or learning time.
     */
    STUDY,

    /**
     * School/class block.
     */
    CLASS,

    /**
     * Meeting, call, appointment.
     */
    MEETING,

    /**
     * Exercise, gym, sport.
     */
    EXERCISE,

    /**
     * Meal time.
     */
    MEAL,

    /**
     * Sleep/rest.
     */
    REST,

    /**
     * Travel/commute.
     */
    TRAVEL,

    /**
     * Personal routine.
     */
    ROUTINE,

    /**
     * Break / free time.
     */
    BREAK,

    /**
     * Work shift or job time.
     */
    WORK,

    /**
     * Flexible placeholder.
     */
    FLEXIBLE,

    /**
     * Unknown or uncategorized.
     */
    OTHER;

    public boolean isFocusBlock() {
        return this == DEEP_WORK
                || this == STUDY
                || this == WORK
                || this == CLASS;
    }

    public boolean isPersonalBlock() {
        return this == EXERCISE
                || this == MEAL
                || this == REST
                || this == ROUTINE
                || this == BREAK;
    }

    public boolean isFixedTimeBlock() {
        return this == CLASS
                || this == MEETING
                || this == WORK
                || this == TRAVEL;
    }

    public boolean isFlexibleBlock() {
        return this == FLEXIBLE || this == OTHER;
    }
}